app/models/projects/acts_as_customizable_patches.rb

```diff
@@ -0,0 +1,248 @@
+      # when we have explicit control over the call of `available_custom_fields`
+      unless global || _query_available_custom_fields_on_global_level
+        custom_fields = custom_fields.where(id: active_custom_field_ids_of_project)
+      end
+
+      custom_fields
+    end
+
+    def available_project_custom_fields_grouped_by_section
+      sorted_available_custom_fields
+        .group_by(&:custom_field_section_id)
+    end
+
+    def available_custom_fields_by_section(section)
+      available_custom_fields
+        .where(custom_field_section_id: section.id)
+    end
+
+    def sorted_available_custom_fields
+      available_custom_fields
+        .sort_by { |pcf| [pcf.project_custom_field_section.position, pcf.position_in_custom_field_section] }
+    end
+
+    def sorted_available_custom_fields_by_section(section)
+      available_custom_fields_by_section(section)
+        .sort_by(&:position_in_custom_field_section)
+    end
+
+    def custom_field_section_ids
+      # we need to check if a project custom field belongs to a specific section when validating
+      # we need a mapping of custom_field_id => custom_field_section_id as we don't want to
+      # change the code of acts_as_customizable for `custom_field_values` which does not include the custom_field_section_id
+      # preloading a hash avoids n+1 queries while validating
+      CustomField
+        .where(id: custom_field_values.pluck(:custom_field_id))
+        .pluck(:id, :custom_field_section_id)
+        .to_h
+    end
+
+    def validate_custom_values
+      # overrides acts_as_customizable
+      # validate custom values only of a specified section
+      # instead of validating ALL custom values like done in acts_as_customizable
+      set_default_values! if new_record?
+
+      custom_field_values
+        .select { |custom_value| of_specified_custom_field_section?(custom_value) }
+        .reject(&:marked_for_destruction?)
+        .select(&:invalid?)
+        .each { |custom_value| add_custom_value_errors! custom_value }
+    end
+
+    def of_specified_custom_field_section?(custom_value)
+      if _limit_custom_fields_validation_to_section_id.present?
+        custom_field_section_ids[custom_value.custom_field_id] == _limit_custom_fields_validation_to_section_id
+      else
+        true # validate all custom values if no specific section was specified
+      end
+    end
+
+    def custom_field_values=(values)
+      # overrides acts_as_customizable
+      # we need to query the available custom fields on a global level when updating custom field values
+      # in order to support implicit activation of custom fields when values are provided during an update
+      self._query_available_custom_fields_on_global_level = true # set to false in after_save hook
+      set_custom_field_values_method_from_acts_as_customizable_module(values)
+    end
```

Comment:
As I tested it, `super` seems to work, combining it with the renamed `with_all_available_custom_fields` this method could be simplified:

```suggestion
    # we need to query the available custom fields on a global level when updating custom field values
    # in order to support implicit activation of custom fields when values are provided during an update
    def custom_field_values=(values)
      with_all_available_custom_fields { super }
    end
```